src/main/scala/is/hail/expr/Type.scala

```diff
@@ -190,6 +187,10 @@ sealed abstract class Type extends Serializable { self =>
   def byteSize: Int = throw new NotImplementedError(toString)
 
   def alignment: Int = byteSize
+
+  /*  Fundamental types are types that can be handled natively by RegionValueBuilder: primitive
+      types, Array and Struct. */
+  def fundamentalType: Type = this
```

Comment:
I did?  Oops.  How were you using it?

We still want them for resolving functions in the function registry, right?  Also, TVariant will soon contain the reference.